### ISSUE TITLE
Question on the README regarding Sarama

### ISSUE THREAD

**Issue by pedromss** [segmentio/kafka-go#377]
On the README you have this sentence:
> sarama, which is by far the most popular but is quite difficult to work with. It is poorly documented, the API exposes low level concepts of the Kafka protocol, and it doesn't support recent Go features like contexts. It also passes all values as pointers which causes large numbers of dynamic memory allocations, more frequent garbage collections, and higher memory usage.

The part I'm focused on is the `all values as pointers which causes large numbers of dynamic memory allocations`. If it passed all values as values wouldn't there be the exact same name of memory allocations but they would mostly be bigger as a pointer would have at most 64 bits (assuming a 64bit arch) whereas a value would have the size of the struct that was being passed?

**Comment by stevevls** [comment]
Hi @pedromss.  I highly recommend you check out a post on our blog that goes into great detail about how allocation works in Go: https://segment.com/blog/allocation-efficiency-in-high-performance-go-services/.  It's a great read, and I think you'll find that it answers your question.

**Comment by QuentinMcgaw-districtm** [comment]
Hello, thanks stevevls, that article is quite interesting.

Would you perhaps be able to point out to code implementation differences between kafka-go and saruma regarding how values are passed? Or at least in what 'hot code path' this is important? Thanks!

Even better would be a memory benchmark between the two libraries but that would be quite time consuming (although valuable for people hesitating between the two libs).

**Comment by stevevls** [comment]
Hi @QuentinMcgaw-districtm.  We don't have time to put together a detailed benchmark at the moment, but I can share this min/avg/max CPU usage graph from a service that we migrated from sarama to kafka-go.  We migrated the canary of this service on day one and then the rest of the tasks the following day.  The only change here was the kafka client library.  GC can be a major drag on high throughput systems!
![image](https://user-images.githubusercontent.com/432860/87339844-0c933c80-c4fc-11ea-826e-fefcc73f1a08.png)
